pex/targets.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os
+
+from pex.interpreter import PythonInterpreter
+from pex.orderedset import OrderedSet
+from pex.pep_425 import CompatibilityTags
+from pex.pep_508 import MarkerEnvironment
+from pex.platforms import Platform
+from pex.third_party.pkg_resources import Requirement
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Iterator, Optional, Tuple
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class Target(object):
```

Comment:
This used to be `DistributionTarget` and was changed from a 2-state single type to two concrete types `LocalInterpreter` and `AbbreviatedPlatform` deriving from this single base class.